Sources/xdm/Application.swift

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright 2021 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+import AEPServices
+import Foundation
+
+struct Application: XDMDirectMappable {
+    init() {}
+
+    var closeType: CloseType?
+    var id: String?
+    var installDate: Date?
+    var isClose: Bool?
+    var isInstall: Bool?
+    var isLaunch: Bool?
+    var isUpgrade: Bool?
+    var name: String?
+    var sessionLength: Int64?
+    var version: String?
+
+    enum CodingKeys: String, CodingKey {
+        case closeType
+        case id
+        case installDate
+        case isClose
+        case isInstall
+        case isLaunch
+        case isUpgrade
+        case name
+        case sessionLength
+        case version
+    }
+
+    static func fromDirect(data: [String: Any]) -> XDMDirectMappable? {
+        let systemInfoService = ServiceProvider.shared.systemInfoService
+        var application = Application()
+    
+        application.id = systemInfoService.getApplicationBundleId()
+        application.name = systemInfoService.getApplicationName()
+        application.version = systemInfoService.getApplicationVersion()
+        // TODO: Close type AMSDK-9271
+        // TODO: isClose AMSDK-9271
+
+        if let lifecycleContextData = data[EdgeConstants.SharedState.Lifecycle.CONTEXT_DATA] as? [String: Any] {
+            application.isInstall = lifecycleContextData[EdgeConstants.SharedState.Lifecycle.ContextData.INSTALL_EVENT] != nil
+            application.isLaunch = lifecycleContextData[EdgeConstants.SharedState.Lifecycle.ContextData.LAUNCH_EVENT] != nil
+            application.isUpgrade = lifecycleContextData[EdgeConstants.SharedState.Lifecycle.ContextData.UPGRADE_EVENT] != nil
```

Comment:
In this case, a hit is not an upgrade this will set the flag to `false` instead of `nil`.  I can update this logic to set to nil if not preset rather than false if we'd prefer to have that behavior for our encoding.